src/fifobuffer.jl

```diff
@@ -79,10 +80,10 @@ Base.read(f::FIFOBuffer) = readavailable(f)
 Base.flush(f::FIFOBuffer) = nothing
 
 function Base.eof(f::FIFOBuffer)
-    return f.eof && f.nb == 0
+    return f.content_length >= 0 && f.content_pos >= f.content_length || f.content_length < 0 && f.nb == 0
 end
 function Base.close(f::FIFOBuffer)
-    f.eof = true
+    f.content_length = f.content_pos
```

Comment:
I think this case is a little more problematic; what if the buffer gets closed before `content_pos` has reached `content_length`?